internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+package bitbucketcloud
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"strings"
+
+	"github.com/ktrysmt/go-bitbucket"
+	"github.com/lindell/multi-gitter/internal/scm"
+	"github.com/pkg/errors"
+)
+
+type BitbucketCloud struct {
+	repositories []string
+	workspaces   []string
+	users        []string
+	username     string
+	token        string
+	sshAuth      bool
+	httpClient   *http.Client
+	bbClient     *bitbucket.Client
+}
+
+func New(username string, token string, repositories []string, workspaces []string, users []string, sshAuth bool, transportMiddleware func(http.RoundTripper) http.RoundTripper) (*BitbucketCloud, error) {
+	if strings.TrimSpace(token) == "" {
+		return nil, errors.New("bearer token is empty")
+	}
+
+	bitbucketCloud := &BitbucketCloud{}
+	bitbucketCloud.repositories = repositories
+	bitbucketCloud.workspaces = workspaces
+	bitbucketCloud.users = users
+	bitbucketCloud.username = username
+	bitbucketCloud.token = token
+	bitbucketCloud.sshAuth = sshAuth
+	bitbucketCloud.httpClient = &http.Client{
+		Transport: transportMiddleware(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: false}, // nolint: gosec
+		}),
+	}
+	bitbucketCloud.bbClient = bitbucket.NewBasicAuth(username, token)
+
+	return bitbucketCloud, nil
+}
+
+func (bbc *BitbucketCloud) CreatePullRequest(ctx context.Context, repo scm.Repository, prRepo scm.Repository, newPR scm.NewPullRequest) (scm.PullRequest, error) {
+
+	splitRepoFullName := strings.Split(prRepo.FullName(), "/")
+
+	repoOptions := &bitbucket.RepositoryOptions{
+		Owner:    bbc.workspaces[0],
+		RepoSlug: splitRepoFullName[1],
+	}
+	defaultReviewers, _ := bbc.bbClient.Repositories.Repository.ListDefaultReviewers(repoOptions)
+	for _, reviewer := range defaultReviewers.DefaultReviewers {
+		newPR.Reviewers = append(newPR.Reviewers, reviewer.Uuid)
+	}
+
+	prOptions := &bitbucket.PullRequestsOptions{
+		Owner:             bbc.workspaces[0],
+		RepoSlug:          splitRepoFullName[1],
+		SourceBranch:      newPR.Head,
+		DestinationBranch: newPR.Base,
+		Title:             newPR.Title,
+		CloseSourceBranch: true,
+		Reviewers:         newPR.Reviewers,
+	}
+
+	_, err := bbc.bbClient.Repositories.PullRequests.Create(prOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pullRequest{
+		project:    splitRepoFullName[0],
+		repoName:   splitRepoFullName[1],
+		branchName: newPR.Head,
+		prProject:  splitRepoFullName[0],
```

Comment:
It seems that `project`/`repoName` and `prProject`/`prRepoName` are used the same here. I guess it comes from other implementations and it might be good to clarify what they are for there.

The first pair are a reference to the project where the PR is place (where is will merge), while the second one is from which repository the change is currently made. If `fork` is not defined, these will be the same, since the changes will happen the same project. But they could be different, and in that case I think this will not be right.